change/@ni-nimble-blazor-803b27ab-c364-4038-afe2-da2ac9153dbf.json

```diff
@@ -0,0 +1,7 @@
+{
```

Comment:
@rajsite @atmgrifter00 Here's the links to the branches I was working on: button-variations (this branch) [button-variations2](https://github.com/ni/nimble/tree/button-variations2) and [button-variations3](https://github.com/ni/nimble/blob/button-variations3/packages/nimble-components/src/patterns/button/styles.ts).

Button-variations currently declares a layer for every css class, which is the functionality we're trying to move away from, but it's worth mentioning as it has correct styling for the buttons.

For button-variations2, I only used the `default` and `disabled` cascade layers. I reorganized the layers and grouped together the css classes that went with default (control, hover, focusVisible, active, etc.), and the ones that went with disabled (just disabled). If you run this in storybook you'll notice a few css styling errors, I intentionally didn't fix these- I'm not adding any extra styling to indicate that using only two cascade layers would require more reiterated css than using multiple.

For button-variations3, I used all 5 of the cascade layers, and grouped each cascade layer with its corresponding css classes (ex. all of the :hover classes went together in the hover layer). I kept all of the classes in the same order within each layer, so they still follow the grouping selectors guidelines. The cascade layers are organized from lowest to highest priority. A few places like `.withBehaviors` and the `buttonAppearanceVariantStyles` required their own layers but there's still significantly less repetition of layer names than there is in this branch (button-variations). The styling in this branch also correctly represents the Figma styling of the button appearance variants.

In terms of styling guidelines, I think that the cascade layers should be organized in their hierarchy, preferably with names that we already use (ex. default / control, hover, focusVisible, active, disabled, ..., then lastly css classes that don't have a layer, as these are the highest priority by default). Within those cascade layers, css classes would follow the [group selectors by target and document order](https://github.com/ni/nimble/blob/main/packages/nimble-components/docs/css-guidelines.md#group-selectors-by-target-and-document-order) guidelines.

Let me know what you think! 😄

Reply:
Between the 3 options (taking into consideration the issue you mentioned for button-variations2), I think my preference is for `button-variations3'. I personally find it helpful to find all of the CSS related to a particular layer contained within a single section.

Reply:
FYI @aidendk @atmgrifter00  prefer using the comment thread on the file tab so conversations can be threaded.

Haven't had a chance to look yet, just moving the thread over

Reply:
@aidendk could you update the PR description to capture where you landed?